from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_Translate(object):

    def show_main(self, main_window, current_window):
        main_window.show()
        current_window.close()

    def setupUi(self, Translate, LessonChoice):
        self.new_word = 1
        Translate.setObjectName("Translate")
        Translate.resize(671, 534)
        Translate.setStyleSheet("background-color: rgb(98, 160, 234);")
        self.centralwidget = QtWidgets.QWidget(Translate)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout.setObjectName("gridLayout")
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.MicrofoneToolButton = QtWidgets.QToolButton(self.centralwidget,
                                                         clicked=lambda: self.show_main(LessonChoice,Translate))
        self.MicrofoneToolButton.setObjectName("MicrofoneToolButton")
        self.verticalLayout.addWidget(self.MicrofoneToolButton)
        spacerItem = QtWidgets.QSpacerItem(25, 9, QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Minimum)
        self.verticalLayout.addItem(spacerItem)
        self.gridLayout.addLayout(self.verticalLayout, 0, 0, 1, 1)
        self.gridLayout_3 = QtWidgets.QGridLayout()
        self.gridLayout_3.setSizeConstraint(QtWidgets.QLayout.SetMinAndMaxSize)
        self.gridLayout_3.setObjectName("gridLayout_3")
        self.fifthWord = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.fifthWord.sizePolicy().hasHeightForWidth())
        self.fifthWord.setSizePolicy(sizePolicy)
        self.fifthWord.setMaximumSize(QtCore.QSize(700, 60))
        self.fifthWord.setStyleSheet("background-color: rgb(229, 165, 10);")
        self.fifthWord.setObjectName("fifthWord")
        self.gridLayout_3.addWidget(self.fifthWord, 9, 0, 1, 1)
        self.fourthWord = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.fourthWord.sizePolicy().hasHeightForWidth())
        self.fourthWord.setSizePolicy(sizePolicy)
        self.fourthWord.setMaximumSize(QtCore.QSize(700, 60))
        self.fourthWord.setStyleSheet("background-color: rgb(245, 194, 17);")
        self.fourthWord.setObjectName("fourthWord")
        self.gridLayout_3.addWidget(self.fourthWord, 8, 0, 1, 1)
        self.firstWord = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.firstWord.sizePolicy().hasHeightForWidth())
        self.firstWord.setSizePolicy(sizePolicy)
        self.firstWord.setMaximumSize(QtCore.QSize(700, 60))
        self.firstWord.setStyleSheet("background-color: rgb(249, 240, 107);")
        self.firstWord.setObjectName("firstWord")
        self.gridLayout_3.addWidget(self.firstWord, 4, 0, 1, 1)
        self.thirdWord = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.thirdWord.sizePolicy().hasHeightForWidth())

        self.thirdWord.setSizePolicy(sizePolicy)
        self.thirdWord.setMaximumSize(QtCore.QSize(700, 60))
        self.thirdWord.setStyleSheet("background-color: rgb(246, 211, 45);")
        self.thirdWord.setObjectName("thirdWord")
        self.gridLayout_3.addWidget(self.thirdWord, 7, 0, 1, 1)
        self.secondWord = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.secondWord.sizePolicy().hasHeightForWidth())
        self.secondWord.setSizePolicy(sizePolicy)
        self.secondWord.setMaximumSize(QtCore.QSize(700, 60))
        self.secondWord.setStyleSheet("background-color: rgb(248, 228, 92);")
        self.secondWord.setObjectName("secondWord")
        self.gridLayout_3.addWidget(self.secondWord, 6, 0, 1, 1)
        spacerItem1 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout_3.addItem(spacerItem1, 10, 0, 1, 1)
        self.gridLayout.addLayout(self.gridLayout_3, 2, 0, 3, 4)
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        spacerItem2 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.horizontalLayout_2.addItem(spacerItem2)
        self.gridLayout.addLayout(self.horizontalLayout_2, 4, 4, 1, 1)
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.word2Translate = QtWidgets.QLabel(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.word2Translate.sizePolicy().hasHeightForWidth())
        self.word2Translate.setSizePolicy(sizePolicy)
        self.word2Translate.setMinimumSize(QtCore.QSize(0, 61))
        self.word2Translate.setStyleSheet("background-color: rgb(153, 193, 241);")
        self.word2Translate.setObjectName("word2Translate")
        self.horizontalLayout_3.addWidget(self.word2Translate)
        self.gridLayout.addLayout(self.horizontalLayout_3, 1, 0, 1, 5)
        self.gridLayout_2 = QtWidgets.QGridLayout()
        self.gridLayout_2.setContentsMargins(-1, 1, -1, -1)
        self.gridLayout_2.setObjectName("gridLayout_2")
        spacerItem3 = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout_2.addItem(spacerItem3, 2, 2, 1, 1)
        self.profileOpenButton = QtWidgets.QPushButton(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.profileOpenButton.sizePolicy().hasHeightForWidth())
        self.profileOpenButton.setSizePolicy(sizePolicy)
        self.profileOpenButton.setMinimumSize(QtCore.QSize(80, 30))
        self.profileOpenButton.setMaximumSize(QtCore.QSize(100, 30))
        self.profileOpenButton.setObjectName("profileOpenButton")
        self.gridLayout_2.addWidget(self.profileOpenButton, 2, 4, 1, 1)
        spacerItem4 = QtWidgets.QSpacerItem(20, 58, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        self.gridLayout_2.addItem(spacerItem4, 2, 0, 1, 1)
        self.start_time = 5
        self.lcdNumber = QtWidgets.QLCDNumber(self.centralwidget)
        self.lcdNumber.setMinimumSize(QtCore.QSize(100, 50))
        self.lcdNumber.setProperty("intValue", 5)
        self.lcdNumber.setObjectName("lcdNumber")
        self.gridLayout_2.addWidget(self.lcdNumber, 2, 1, 1, 1)
        self.gridLayout.addLayout(self.gridLayout_2, 0, 1, 1, 5)
        self.graphicsView = QtWidgets.QGraphicsView(self.centralwidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.graphicsView.sizePolicy().hasHeightForWidth())
        self.graphicsView.setSizePolicy(sizePolicy)
        self.graphicsView.setMaximumSize(QtCore.QSize(100, 100))
        self.graphicsView.setObjectName("graphicsView")
        self.gridLayout.addWidget(self.graphicsView, 1, 5, 1, 1)

        Translate.setCentralWidget(self.centralwidget)

        self.retranslateUi(Translate)
        QtCore.QMetaObject.connectSlotsByName(Translate)

        self.timer = QtCore.QTimer()
        self.timer.timeout.connect(self.updateLCD)
        self.timer.start(1000)

    def timer_lcd(self):
        if self.start_time != 0:
            self.start_time -= 1
        else:
            self.new_word += 1
            self.fourthWord.setText(str(self.new_word))
            self.word2Translate.setStyleSheet("background-color: red")
            self.timer.stop()

    def updateLCD(self):
        self.timer_lcd()
        self.lcdNumber.display(self.start_time)




    def retranslateUi(self, Translate):
        _translate = QtCore.QCoreApplication.translate
        Translate.setWindowTitle(_translate("Translate", "Translate"))
        self.MicrofoneToolButton.setText(_translate("Translate", "..."))
        self.fifthWord.setText(_translate("Translate", str(self.new_word)))
        self.fourthWord.setText(_translate("Translate", "fourth_word"))
        self.firstWord.setText(_translate("Translate", "first_word"))
        self.thirdWord.setText(_translate("Translate", "third_word"))
        self.secondWord.setText(_translate("Translate", "second_word"))
        self.word2Translate.setText(_translate("Translate", "<html><head/><body><p align=\"center\">TextLabel</p></body></html>"))
        self.profileOpenButton.setText(_translate("Translate", "Profile"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Translate = QtWidgets.QMainWindow()
    ui = Ui_Translate()
    ui.setupUi(Translate)
    Translate.show()
    sys.exit(app.exec_())
